eng/pipelines/templates/jobs/build-cli.yml

```diff
@@ -88,14 +88,25 @@ jobs:
       - bash: dotnet nuget add source --name dotnet9 https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json
         displayName: Add internal dotnet nuget feed
 
-      - template: /eng/pipelines/templates/steps/azd-login.yml
-        parameters:
-          AzdDirectory: cli/azd
+      - template: /eng/pipelines/templates/steps/configure-oidc-auth.yml
+
+      - pwsh: |
+            ./azd auth login --federated-credential-provider "azure-pipelines"
+            ./azd config set defaults.subscription $(SubscriptionId)
+        condition: and(succeeded(), ne(variables['Skip.LiveTest'], 'true'))
+        workingDirectory: cli/azd
+        env:
+          AZURESUBSCRIPTION_CLIENT_ID: $(AzureSubscriptionClientId)
+          AZURESUBSCRIPTION_TENANT_ID: $(AzureSubscriptionTenantId)
+          AZURESUBSCRIPTION_SERVICE_CONNECTION_ID: $(AzureSubscriptionServiceConnectionId)
+          SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+          AZD_DEBUG_LOGIN_FORCE_SUBSCRIPTION_REFRESH: true
+        displayName: AZD Login
 
       - task: AzureCLI@2
         condition: and(succeeded(), ne(variables['Skip.LiveTest'], 'true'))
         inputs:
-          azureSubscription: azure-sdk-tests
+          azureSubscription: azd-service-connection
```

Comment:
Are terraform tests running in this context? If yes, no action required here. 

If no... 

`AzureCLI@2` should not be necessary here given that everything needed for auth is set in `configure-oidc-auth.yml`. Replacing this with a `PowerShell@2` (and no service connection) could take up to a minute off this task because we won't need to do any of the `az login` work that `AzureCLI@2` does (that's all handled by `configure-oidc-auth.yml`)

Reply:
Looks like terraform tests are skipped: 

![image](https://github.com/user-attachments/assets/3096851a-d3dd-4042-b4d4-c07ee21277e9)

Reply:
I want to get these terraform tests working again shortly, so I will leave this as is for now.